synapse/__init__.py

```diff
@@ -61,11 +61,20 @@
 except ImportError:
     pass
 
-# Use the standard library json implementation instead of simplejson.
+# Teach canonicaljson how to serialise immutabledicts.
 try:
-    from canonicaljson import set_json_library
-
-    set_json_library(json)
+    from canonicaljson import register_preserialisation_callback
```

Comment:
The comment above says that we need to ensure this file is executable so that we can read `__version__` from it via setup.py. The version is now declared in pyproject.toml, so I think we could clean this up here.

Reply:
> I think we could clean this up here

Will leave that for the future---this already touched a lot of files.